src/org/exist/http/servlets/XSLTServlet.java

```diff
@@ -348,26 +337,7 @@ private Templates getSource(Subject user, HttpServletRequest request, HttpServle
             }
         }
 
-        // Try to figure out the base directory of the stylesheet file.
-        // This is required to locate resources imported within the stylesheet.
-        String base;
-        final int p = stylesheet.lastIndexOf("/");
-        if (p != Constants.STRING_NOT_FOUND) {
-            base = stylesheet.substring(0, p);
-        } else {
-            base = stylesheet;
-        }
-
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("Loading stylesheet from " + stylesheet);
-        }
-
-        CachedStylesheet cached = cache.get(stylesheet);
-        if (cached == null) {
-            cached = new CachedStylesheet(factory, user, stylesheet, base);
-            cache.put(stylesheet, cached);
-        }
-        return cached.getTemplates(user);
+        return TemplatesFactory.stylesheet(stylesheet, "", isCaching());
```

Comment:
Just put a check in the top of `TemplatesFactory#stylesheet` which checks if null was supplied and uses an empty string. Passing an empty string seems like it will lead to mistakes in future.